### DISCUSSION TITLE
Can't download the script - python-pip Problems

### DISCUSSION THREAD

**Discussion by idscomm** [ep1cman/unifi-protect-backup#52] [Q&A]
Hello,

I have been trying to install this great tool you created for Unifi Protect but I can't get the script to work...  I am running Ubuntu 18 with python 3.9 and pip 21 ... This is what I get when I type "pip install unifi-protect-backup"

> ERROR: Could not find a version that satisfies the requirement unifi-protect-backup (from versions: none)
> ERROR: No matching distribution found for unifi-protect-backup

I also tried "python3.9 -m pip install unifi-protect-backup" but I got this error:

> Traceback (most recent call last):
>   File "/usr/lib/python3.9/runpy.py", line 188, in _run_module_as_main
>     mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
>   File "/usr/lib/python3.9/runpy.py", line 147, in _get_module_details
>     return _get_module_details(pkg_main_name, error)
>   File "/usr/lib/python3.9/runpy.py", line 111, in _get_module_details
>     __import__(pkg_name)
>   File "/usr/lib/python3/dist-packages/pip/__init__.py", line 29, in <module>
>     from pip.utils import get_installed_distributions, get_prog
>   File "/usr/lib/python3/dist-packages/pip/utils/__init__.py", line 23, in <module>
>     from pip.locations import (
>   File "/usr/lib/python3/dist-packages/pip/locations.py", line 9, in <module>
>     from distutils import sysconfig
> ImportError: cannot import name 'sysconfig' from 'distutils' (/usr/lib/python3.9/distutils/__init__.py)

Any idea where to start? Any help would be appreciated!

Thanks!

**Comment by ep1cman** [comment]
Thank you for your patience waiting for a response. I did some digging and it looks like ubuntu might have an issue with the system distutils: https://stackoverflow.com/a/64771661

I would always recommend running a virtual environment (venv) to help try to mitigate weird issues like this with system python versions. [`pipx`](https://github.com/pypa/pipx) lets you install python tools system wide in their own venvs. 

Alternatively, you can also try docker, that is the way I would recommend using the tools since everything is self contained then.
